x-pack/plugins/search_playground/public/hooks/use_ai_assist_chat.ts

```diff
@@ -184,7 +184,7 @@ export function useAIAssistChat({
       if (messagesRef.current.length === 0) return null;
 
       const chatRequest: ChatRequest = {
-        messages: messagesRef.current,
+        messages: messagesRef.current.slice(0, messagesRef.current.length-1),
```

Comment:
could you double check what happens with errors. 

Example:
- send a message
- get AI response
- update Gen ai API key to be invalid
- send a message
- click re-generate 

I think you need to remove all recent conversations upto the last human message. This means discarding all system or assistant messages after the last human message.